FrostingContextExtensions.cs

```diff
@@ -0,0 +1,27 @@
+namespace BuildScripts;
+
+public static class FrostingContextExtensions
+{
+    private static readonly ProcessSettings _processSettings;
+
+    static FrostingContextExtensions() => _processSettings = new ProcessSettings();
+
+    public static void SetShellWorkingDir(this FrostingContext context, string path)
+    {
+        _processSettings.WorkingDirectory = path;
+    }
+
+    public static int ShellExecute(this FrostingContext context, string command, string environmentVariables = "")

```

Comment:
Pass env variables in a separate function `SetEnvVariables(parms string[] env)`

Reply:
Done in [23d2fc1](https://github.com/MonoGame/MonoGame.Tool.BuildScripts/pull/4/commits/23d2fc12a44a3493c071aa9c863ae82dfe29c693)

From our previous discussion, the `export` keyword is in fact required.  Without it, the environment variables do not stay set if the command executes something that in turn executes something else.  So the `export` is required for the entire run of the process.

Reply:
Additionally, I was not able to find any decent way of doing space character escapes in the command.  The problem comes in situations where you want to set an environment variable such as

```cs
context.SetShellEnvironmentVariables("CFLAGS", $"-w -arch arm64 -I{dependencyDir}/include");
```

The space escapes would need to be set for the path in the `-I` flag, however you would need to differentiate between what is a path and what isn't, such as the `-w -arch arm64` part.   Otherwise, you end up with a sanitized environment variable such as `CFLAGS="-w\ -arch\ arm64\ -i/path/to\ some/directory/include;"` which fails to run.

Open to suggestions on this if still needed

Reply:
You are supposed to set it to either:
```
CFLAGS=-w\ -arch\ arm64\ -i/path/to\ some/directory/include;
```
or
```
CFLAGS="-w -arch arm64 -i/path/to some/directory/include;"
```